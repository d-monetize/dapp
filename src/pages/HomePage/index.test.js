import React from "react"
import { shallow } from "enzyme"
import { HomePage } from "./index"

const props = {
  history: {
    push: jest.fn(),
  },
}

beforeEach(() => {
  props.history.push.mockClear()
})

test("it renders", () => {
  const component = shallow(<HomePage {...props} />)

  expect(component).toMatchSnapshot()
})

test("onSubmit", () => {
  const component = shallow(<HomePage {...props} />)

  component.find("SearchSubscriptionForm").simulate("submit", "address")

  expect(props.history.push.mock.calls.length).toEqual(1)
  expect(props.history.push.mock.calls[0][0]).toEqual("/subscriptions/address")
})

test("onClickCreateSubscription", () => {
  const component = shallow(<HomePage {...props} />)

  component.find("Button").simulate("click")

  expect(props.history.push.mock.calls.length).toEqual(1)
  expect(props.history.push.mock.calls[0][0]).toEqual("/subscriptions/create")
})
