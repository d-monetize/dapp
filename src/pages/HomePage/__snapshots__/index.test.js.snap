// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
<div
  className="HomePage"
>
  <div
    className="main"
  >
    <div
      className="logo"
    >
      <h1
        className="header"
      >
        D Monetize
      </h1>
    </div>
    <div
      className="form"
    >
      <SearchSubscriptionForm
        onSubmit={[Function]}
      />
    </div>
    <Button
      as="button"
      color="green"
      onClick={[Function]}
      size="huge"
    >
      Create Subscription
    </Button>
  </div>
</div>
`;
